api/desecapi/serializers.py

```diff
@@ -793,17 +793,43 @@ def save(self, **kwargs):
         raise ValueError
 
 
-class AuthenticatedBasicUserActionSerializer(AuthenticatedActionSerializer):
+class AuthenticatedBasicUserActionMixin():
+    def save(self, recipient=None):
+        context = {**self.context, 'action_serializer': self}
+        return self.action_user.send_email(self.reason, context=context, recipient=recipient)
```

Comment:
consider moving this to models.py

Reply:
I thought about this, and I don't see how it could be done:

- `send_email` is already in models.py (belongs to `User`)
- `save()` has to remain some way or the other, as it is used for the part of the serializer that changes state (here: sends email). See also https://github.com/desec-io/desec-stack/pull/591#issuecomment-1123577349 and https://github.com/desec-io/desec-stack/pull/591#issuecomment-1163349605
- What remains is that `save()` could be calling `save()` on the corresponding `AuthenticatedAction` instance, which would have to be instantiated first. -- But with which arguments? (I'm not sure if `self.Meta.model(**self.data).save()` would always be correct.)
- What's more, sending the email in `AuthenticatedAction.save()` wouldn't work when we have a `ListSerializer` (e.g. for multiple domain renewals), where we only want to send one email (but there would be multiple `AuthenticatedAction` instances -- if we call `save()` on all of them, we'd get multiple emails).

All in all, I think it's best to leave this as is. It seems rather natural that the serializer coalesces any `many=True` actions, and then does what ever to initiate the state change.